doschema/merger.py

```diff
@@ -0,0 +1,193 @@
+                self.process_type_field(fields_types_dict, field_value)
+            elif field_name == 'items':
+                if isinstance(field_value, list):
+                    self.process_array(fields_types_dict, field_value)
+
+    def process_array(self, fields_types_dict, field_value):
+        """Process each json object in the array according to ignore_index.
+
+        :param fields_types_dict: Dictionary keeping path and type of
+                                  the field.
+        :param field_value: Type of the field.
+        """
+        field_path = self.curr_field + ('items', )
+
+        for elem in field_value:
+            if self.ignore_index:
+                new_field_path = field_path
+            else:
+                new_field_path = field_path + (field_value.index(elem), )
+
+            JsonSchemaCompatibilityChecker(
+                self.schema_id,
+                new_field_path,
+                elem,
+                self.ignore_index
+            ).traverse(fields_types_dict)
+
+    def process_type_field(self, fields_types_dict, field_value):
+        """Check if a name field with type "type" has already appeared.
+
+        If had appeared with different type,raise an exception.
+        Otherwise, add object to a dict of types.
+
+        :param fields_types_dict: Dictionary keeping path and type of
+                                  the field.
+        :param field_name: Name of the field.
+        :param field_value: Type of the field.
+        """
+        if self.curr_field in fields_types_dict.keys():
+            if fields_types_dict[self.curr_field].field_type != field_value:
+                raise JsonSchemaCompatibilityException({
+                    'field_name': self.curr_field,
+                    'old_type': fields_types_dict[
+                        self.curr_field
+                    ].field_type,
+                    'old_schema': fields_types_dict[
+                        self.curr_field
+                    ].schema_index,
+                    'new_type': field_value,
+                    'new_schema': self.schema_id
+                })
+        else:
+            fields_types_dict[self.curr_field] = FieldToAdd(
+                self.schema_id,
+                self.curr_field,
+                field_value
+            )
+
+    def collect_special(self, fields_types_dict, json_objects_list):
+        """Operate a field with name 'allOf', 'anyOf' or 'oneOf'.
+
+        :param fields_types_dict: Dictionary keeping path and type of
+                                  the field.
+        :param json_objects_list: List of json objects.
+        """
+        for elem in json_objects_list:
+            JsonSchemaCompatibilityChecker(
+                self.schema_id,
+                self.curr_field,
+                elem,
+                self.ignore_index
+            ).traverse(fields_types_dict)
+
+    @staticmethod
+    def starter(schemas_list, ignore_index=True):
```

Comment:
functions should have a verb as name. They are performing an action.

Also this can be moved to a separate function as it does not depend in the class nor on a specific instance. Users don't get anything from knowing that your implementation is using a class.

This function is the only thing useful to the outside world here, the class name `JsonSchemaCompatibilityChecker ` should start with an `_` so that it is not imported by default.